tests/test_api.cpp

```diff
@@ -242,6 +242,91 @@ TEST( ApiTest, VerticesApi )
   EXPECT_EQ( MDAL_FI_next( nullptr, 0, nullptr, 0, nullptr ), 0 );
 }
 
+void _populateEdges( MDAL_MeshH m, std::vector<int> &start, std::vector<int> &stop, size_t itemsLen )
+{
+  int edgesCount = MDAL_M_edgeCount( m );
+  start.resize( 0 );
+  stop.resize( 0 );
+  std::vector<int> buffer1( itemsLen );
+  std::vector<int> buffer2( itemsLen );
+
+  MDAL_MeshEdgeIteratorH it = MDAL_M_edgeIterator( m );
+  int edgeIndex = 0;
+  while ( edgeIndex < edgesCount )
+  {
+    int edgesRead = MDAL_EI_next( it,
+                                  static_cast<int>( itemsLen ),
+                                  buffer1.data(),
+                                  buffer2.data() );
+    if ( edgesRead == 0 )
+      break;
+
+    ASSERT_TRUE( edgesRead <= static_cast<int>( itemsLen ) );
+
+    start.insert( start.end(),
+                  buffer1.begin(),
+                  buffer1.begin() + edgesRead );
+
+    stop.insert( stop.end(),
+                 buffer1.begin(),
+                 buffer1.begin() + edgesRead );
+
+    edgeIndex += edgesRead;
+  }
+  MDAL_EI_close( it );
+}
+
+TEST( ApiTest, EdgesApi )
+{
+  std::string path = test_file( "/ply/all_features.ply" );
+  MDAL_MeshH m = MDAL_LoadMesh( path.c_str() );
+  EXPECT_NE( m, nullptr );
+  MDAL_Status s = MDAL_LastStatus();
+  ASSERT_EQ( MDAL_Status::None, s );
+
+  // reference buffer where taken in one go
+  std::vector<int> refStart;
+  std::vector<int> refStop;
+  _populateEdges(
+    m,
+    refStart,
+    refStop,
+    static_cast<size_t>( MDAL_M_edgeCount( m ) )
+  );
+
+  std::vector<int> start;
+  std::vector<int> stop;
+  {
+    std::vector<int> start;
+    std::vector<int> stop;
+    _populateEdges( m,
+                    start,
+                    stop,
+                    10
+                  );
+
+    EXPECT_TRUE( compareVectors( refStart, start ) );
+    EXPECT_TRUE( compareVectors( refStop, stop ) );
+  }
+
+  {
+    std::vector<double> coords;
+    _populateEdges( m,
+                    start,
+                    stop,
+                    10000
+                  );
+
+    EXPECT_TRUE( compareVectors( refStart, start ) );
+    EXPECT_TRUE( compareVectors( refStop, stop ) );
+  }
+  MDAL_CloseMesh( m );
+
+  // Some wrong calls tests
+  EXPECT_EQ( MDAL_M_edgeIterator( nullptr ), nullptr );
+  EXPECT_EQ( MDAL_EI_next( nullptr, 0, nullptr, nullptr ), 0 );
```

Comment:
should be separate test case and not mixed with regular tests

Reply:
Not totally sure which part you are referring to and what you mean.

Reply:
just that I would not mix in on test function both tests that check that the api is working well with invalid inputs e.g. `EXPECT_EQ( MDAL_M_edgeIterator( nullptr ), nullptr );` and a normal test case where you test the api works for regular use case.

Reply:
but it is minor, do not worry